src/Lykke.Service.EasyBuy.DomainServices/OrdersService.cs

```diff
@@ -0,0 +1,190 @@
+        private readonly ISettingsService _settingsService;
+        private readonly IInternalTransfersService _internalTransfersService;
+        private readonly IPricesService _pricesService;
+        private readonly IOrdersRepository _ordersRepository;
+        private readonly ITradesRepository _tradesRepository;
+        private readonly ILog _log;
+
+        public OrdersService(
+            IPricesService pricesService,
+            IInstrumentsAccessService instrumentsAccessService,
+            IInternalTransfersService internalTransfersService,
+            IAssetsServiceWithCache assetsService,
+            IOrdersRepository ordersRepository,
+            ISettingsService settingsService,
+            ITradesRepository tradesRepository,
+            ILogFactory logFactory)
+        {
+            _pricesService = pricesService;
+            _internalTransfersService = internalTransfersService;
+            _assetsService = assetsService;
+            _ordersRepository = ordersRepository;
+            _settingsService = settingsService;
+            _tradesRepository = tradesRepository;
+            _instrumentsAccessService = instrumentsAccessService;
+            _log = logFactory.CreateLog(this);
+        }
+        
+        public async Task<Order> CreateAsync(string walletId, string priceId, decimal quotingVolume)
+        {
+            var price = await _pricesService.GetSnapshotAsync(priceId);
+            
+            if(price == null)
+                throw new EntityNotFoundException();
+
+            var instrument = await _instrumentsAccessService.GetByAssetPairIdAsync(price.AssetPair);
+            
+            if(DateTime.UtcNow > price.ValidTo + (instrument.AllowedOverlap ?? await _settingsService.GetDefaultAllowedOverlapAsync()))
+                throw new OperationFailedException("Given price too old.");
+            
+            if(quotingVolume > price.QuotingVolume)
+                throw new OperationFailedException("Requested volume higher than initial.");
+
+            var pair = await _assetsService.TryGetAssetPairAsync(price.AssetPair);
+
+            var baseAsset = await _assetsService.TryGetAssetAsync(pair.BaseAssetId);
+
+            var baseVolume = quotingVolume == price.QuotingVolume
+                ? price.BaseVolume
+                : (quotingVolume / price.Price).TruncateDecimalPlaces(baseAsset.Accuracy, price.Type == OrderType.Sell);
+            
+            var order = new Order
+            {
+                Id = Guid.NewGuid().ToString(),
+                WalletId = walletId,
+                Type = price.Type,
+                AssetPair = price.AssetPair,
+                QuotingVolume = quotingVolume,
+                BaseVolume = baseVolume,
+                PriceId = priceId,
+                CreatedTime = DateTime.UtcNow,
+                Status = OrderStatus.New,
+                ReserveTransferId = Guid.NewGuid().ToString(),
```

Comment:
These properties should not be initialized. When transaction ID received from ME these properties can be set.